server/backends/blobstore/aws/aws.go

```diff
@@ -178,8 +178,17 @@ func (a *AwsS3BlobStore) createBucketIfNotExists(ctx context.Context, bucketName
 func (a *AwsS3BlobStore) ReadBlob(ctx context.Context, blobName string) ([]byte, error) {
 	start := time.Now()
 	b, err := a.download(ctx, blobName)
-	util.RecordReadMetrics(awsS3Label, start, b, err)
-	return util.Decompress(b, err)
+	duration := time.Since(start)
+	// We update the err later on, so we need to wrap RecordReadMetrics in a func.
+	defer func() {
+		util.RecordReadMetrics(awsS3Label, duration, int64(len(b)), err)
```

Comment:
If you used a name return you don't need to worry about how err is set.
You can write `func (a *AwsS3BlobStore) ReadBlob(ctx context.Context, blobName string) (rb []byte, retErr error)`

and then in your defer you can reference retErr which will always contain the returned error

if you do that, you can return `util.Decompress(rd)` directly without storing the error.